scopes.md

```diff
@@ -0,0 +1,138 @@
+
+### interactions
+
+This scope gives access to the following requests:
+- Create/delete comments
+- Report posts/comments
+- Post/remove likes
+- Subscribe/mute posts
+- Hide posts from streams
+
+### openid
+
+This scope is defined by the OpenID Connect spec. This is a mandatory scope. It gives access to OpenID Connect authentication.
+
+### private
+
+This scope gives access to the following requests:
+- Read private posts
+- Read private posts intercations (likes, comments)
+
+Reading private posts of contacts additionaly requires `contacts:read` scope.
+
+### public
+
+This scope gives access to the following requests:
+- Read public profiles of users
+- Perform search among publicly searchable profiles
+- Read public posts
+- Read reshares
+- Read public posts intercations (likes, comments)
+
+### posts:publish
+
+This scope gives access to the following requests:
+- Create posts
+- Create reshares
+- Post photos
+
+Create private posts additionaly requires `contacts:read` scope.
+
+### posts:modify
+
+This scope gives access to the following requests:
+- Delete posts
+- Delete photos
+
+### profile
+
+This scope is defined by the OpenID Connect spec. It allows reading user claims (user properties). Claims are read using UserInfo route or from ID Token.
+
+User claims defined by the OpenID Connect are somewhat similar to users own profile data. Generally this scope means "Allow reading user's own profile". Besides user claims defined by the standard this scope also gives access to "read own profile" diaspora* API endpoint.
+
+Supported claims by diaspora:
+
+| claim      | meaning                                             |
+| ---------- | --------------------------------------------------- |
+| name       | Full name                                           |
+| nickname   | The left part of diaspora handle before `@` sign    |
+| profile    | User's profile URL address                          |
+| picture    | User avatar URL address                             |
+| gender     | Gender profile field                                |
+| birthdate  | Birth date profile field                            |
+| locale     | UI language selection specified in account settings |
+| updated_at | Date of last profile edit                           |
+
+### profile:modify
+
+This scope gives access to updating users own profile
+
+### tags
+
+This scope gives access to read/create/delete tag followings.
+
+### streams
+
+This scope gives access streams.
+
+There are 3 types of streams available. Accessing each stream requires two scopes: `streams` scope and a scope specific for the stream type.
+
+| Stream name     | Stream specifics                           | Access scopes               |
+| --------------- | ------------------------------------------ | --------------------------- |
+| Activity stream | Includes posts with users own activity     | `streams` + `interactions`  |
```

Comment:
Similarly, I might want to grant access to reading interactions but not actually modifying any of my data

Reply:
Also the same thing should apply to other streams. Aspects and tags stream contents should depend on `public` and `private` scopes too. Depending on scopes there will be different contents returned by stream endpoints.

Aspects stream requires `contacts:read` and one of `public` and `private`. Activity stream and tags stream require one of `public` and `private` and with `contacts:read` the response includes your contacts private posts (and not only your posts as with just `private`).

Reply:
I'm not entirely sure the public/private separation is really necessary, that is accessing public only content would be a feature that's ok to leave out for now to me or only have available as a separate endpoint with all public content.

@HankG Any thoughts on this from your side?

Reply:
However above in the document we state that `public` is required to access public posts and `private` is required to access private posts. Technically it means that we can grant access to private posts only or public posts only. If we don't want this public/private only access then we probably should redefine these `public` and `private` scopes too. For instance we could define a scope for accessing posts (public and your own private) and which will also grant access to your friends posts when combined with `contacts:read`.